packages/klighd-core/src/views-rendering.tsx

```diff
@@ -266,40 +266,195 @@ export function renderLine(rendering: KPolyline,
+                    } else {
+                        break
+                    }
+                }
+
+                // determine new start point
+                let start_choice = scaled_start
+
+                if (i < points.length) {
+
+                    const remainingPoints = points.length - i
+                    const z = Math.min(max_coord_per_point, remainingPoints)
+
+                    const prev = points[i-1]
+                    const next = points[i + z - 1]
+
+                    const edge = new PointToPointLine(prev, next)
+
+                    const intersections = ScalingUtil.intersections(s_scaled.bounds, edge)
+
+                    intersections.sort(ScalingUtil.sort_by_dist(next))
+
+                    if (intersections.length > 0){
+                        start_choice = intersections[0]
+                    }
+
+                }
+
+                newPoints.push(start_choice)
+
+
+                // keep points not in end node
+                while (i < points.length) {
+                    const remainingPoints = points.length - i
+
+                    const z = Math.min(max_coord_per_point, remainingPoints)
+
+                    const p = points[i + z - 1]
+
+                    if (
+                        !Bounds.includes(t_scaled.bounds, p)
+                    ) {
+                        for (let j = 0; j < z ; j++) {
+                            newPoints.push(points[i])
+                            i++
+                        }
+                    } else {
+                        break
+                    }
+                }
+
+                // determine new end point
+
+                let end_choice = scaled_end
+                if (i < points.length) {
+
+                    const remainingPoints = points.length - i
+                    const z = Math.min(max_coord_per_point, remainingPoints)
+
+                    const prev = points[i-1]
+                    const next = points[i + z - 1]
+
+                    const edge = new PointToPointLine(prev, next)
+
+                    const intersections = ScalingUtil.intersections(t_scaled.bounds, edge)
+
+                    intersections.sort(ScalingUtil.sort_by_dist(prev))
+
+                    if (intersections.length > 0){
+                        end_choice = intersections[0]
+                    }
+
+                    // keep the control points of the current point
+                    if (z >= 2) {
+                        newPoints.push(points[i])
+                        if (z == 3) {
+                            newPoints.push(points[i + 1])
+                        }
+                    }
+
+                }
+
+                newPoints.push(end_choice)
+
+                parent.moved_ends_by = {start: Point.subtract(start_choice, points[0]), end: Point.subtract(end_choice, points[points.length -1])}
+                points = newPoints
+                break;
+            }
+            case K_POLYGON: {
```

Comment:
This case handles the arrow heads of edges, it 
1. applies the transformations as already calculated (`gAttrs.transform`) 
2. translates the polygon so that the tip should point at the origin
3. applies up-scaling 
4. translates to the polygon to point the tip at the adjusted end of the edge

Reply:
added some comments